tensorflow/python/BUILD

```diff
@@ -4055,7 +4055,23 @@ py_library(
 py_library(
     name = "special_math_ops",
     srcs = ["ops/special_math_ops.py"],
-    srcs_version = "PY2AND3",
+    srcs_version = "PY3ONLY",
+    deps = [
+        ":array_ops",
+        ":check_ops",
+        ":control_flow_ops",
+        ":framework_for_generated_wrappers",
+        ":math_ops",
+        ":platform",
+        "//tensorflow/compiler/tf2xla/ops:gen_xla_ops",
+        "@opt_einsum_archive//:opt_einsum",
+    ],
+)
+
+py_library(
+    name = "special_math_ops",
```

Comment:
You cannot have two targets with the same name. Why not leave this as `PY2AND3` instead of creating 2 targets, one `PY3ONLY` and one `PY2ONLY`?

Reply:
The problem is that the unit tests fail. This is because during the bazel build for python3, the deps list of special_math_ops contains functools32 as a dependency, which now is a PY2ONLY package. It then complains about needing to install a py2 package. Somehow, I need to let bazel know that the special_math_ops lib should be build with different dependencies on python2 and python3. I'm not a bazel expert, but I have not found any way to do this yet, and I guess this is reason why functools32 was setup to be a PY2AND3 library in the first place. Seeing as a fix for not using it in python3 is hacked in later in the setup.py. If this is not doable in bazel, I think that means there needs to be a entire separate py_library setup for python2 and 3, which I think is not something I should get involved with but rather some architect.

Only other option I saw online was to do something like this:
`"@functools32_archive//:functools32/python:PY2"`
But I cannot see any official docs mentioning this to even be possible